java/vector/src/main/java/org/apache/arrow/vector/util/FieldVectorUtility.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector.util;
+
+import io.netty.buffer.ArrowBuf;
+import org.apache.arrow.vector.BaseFixedWidthVector;
+import org.apache.arrow.vector.BaseVariableWidthVector;
+import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.vector.ipc.ReadChannel;
+import org.apache.arrow.vector.ipc.WriteChannel;
+import org.apache.arrow.vector.ipc.message.MessageSerializer;
+
+import java.io.IOException;
+
+/**
+ * Utility class for FieldVector.
+ */
+public class FieldVectorUtility {
+
+  public static void serializeFieldVector(WriteChannel out, FieldVector vector) throws IOException {
+    int valueCount = vector.getValueCount();
+    byte[] reuseIntBytes = new byte[4];
+    MessageSerializer.intToBytes(valueCount, reuseIntBytes);
+    out.write(reuseIntBytes);
+    ArrowBuf dataBuffer = vector.getDataBuffer();
+    ArrowBuf validityBuffer = vector.getValidityBuffer();
+
+    int validityBufferCapacity = (valueCount + 7) >> 3;
```

Comment:
there are already utility methods that have this implementation can you use one of them?